### ISSUE TITLE
SingleTaskGP versus MultiTaskGP

### ISSUE THREAD

**Issue by madhavkrishnan94** [pytorch/botorch#1761]
## Issue description

Hello!

We are currently using botorch to train a multi-output GP model on our data. Let's say, the GP model is trying to fit the function _f_ on our dataset [_Y=f(X)_], where _Y_ is a 4-dimensional vector of the output, i.e., _Y = [Y1, Y2, Y3, Y4]_. Similarly, _X_ is a 3-dimensional vector of the input, i.e., _X = [X1, X2, X3]_.

We also note that the output data is correlated.

In this regard, we have two questions:

1. In the [documentation](https://botorch.org/api/models.html#module-botorch.models.gp_regression), it says:

> However, as single-task models, SingleTaskGP, FixedNoiseGP, and HeteroskedasticSingleTaskGP should be used only when the outputs are independent and all use the same training data. If outputs are independent and outputs have different training data, use the ModelListGP. When modeling correlations between outputs, use a multi-task model like MultiTaskGP.

We would like to know what "different training data" here actually means. 

2. At the outset, we have a single-task multi-output problem and using `SingleTaskGP` makes more sense (going by the documentation). However, `SingleTaskGP` does not capture the correlations between outputs. As a result, we would like to check if the following approach is feasible-

We would like to use `MultiTaskGP` by predicting each output in a different task (hence 4 tasks). By doing this, we would be able to leverage upon the ability of `MultiTaskGP` to capture correlation across tasks, thereby capturing correlation between outputs.

____________________________

We would like to hear the thoughts of the botorch community on these questions. Thanks!

**Comment by Balandat** [comment]
> We would like to know what "different training data" here actually means.

By that we mean that the observations are not of what is sometimes called a "block design", where each output is observed at the same location - i.e. if you evaluate `X`, you always get all of the `Y`s and not just some (which seems to be the case in your setting?). 

> We would like to use MultiTaskGP by predicting each output in a different task (hence 4 tasks). By doing this, we would be able to leverage upon the ability of MultiTaskGP to capture correlation across tasks, thereby capturing correlation between outputs.

Yes, that makes sense! Are your observations noisy? Or are they deterministic? If they are deterministic AND your data has the block design described above, there isn't actually any benefit from modeling this with a multi-task model (this is a property called autokrigeability).

**Comment by madhavkrishnan94** [comment]
Thanks a lot for your reply.

In my case, the observations are deterministic. And they have the block design, i.e., for each `X`, all `Y` s are obtained.

So far, I have been using `SingleTaskGP` to infer the intrinsic noise.

**Comment by Balandat** [comment]
> So far, I have been using SingleTaskGP to infer the intrinsic noise.

Hmm if the observations are deterministic, why infer a noise (that you know is zero)? You can instead use a `FixedNoiseGP` passing zero variance (under the hood we'll convert this to some small jitter value for numerical stability). 

Again, if you have deterministic observations AND your observations are of a block design, then there is no need to use a multi-task model (that is computationally a lot more expensive).

**Comment by madhavkrishnan94** [comment]
Thanks a lot for your inputs.

**Comment by Balandat** [comment]
You won't get any benefit in this case from capturing the correlation due to the autokrigeability.

**Comment by madhavkrishnan94** [comment]
Thank you.

And I wanted to clarify one more thing wrt the observations being deterministic as I believe I misunderstood the context of the term. We actually have noise in the `Y` data; it contains 3 repeats on the same `X`. So, I guess using `SingleTaskGP` is more suitable in inferring the intrinsic noise.

**Comment by esantorella** [comment]
Yes, if you have multiple results on the same `X` with different values of `Y`, you have noise. Assuming you don't know the level of noise, it would be appropriate to use `SingleTaskGP` or `HeteroskedasticSingleTaskGP` rather than `FixedNoiseGP` so that the model can infer the noise.
